### PR TITLE
fix(usdt-approval): :bug: fix modal not dismissing when close is clicked

### LINKED ISSUE #2522: [1.38] USDT approval flow: not possible to close the reset allowance modal
# To Reproduce

1. Go to CoW Swap on Goerli,
2. Use USDT https://goerli.etherscan.io/address/0x7b77F953e703E80CD97F6911385c0b1ceabC96Bc
3. Approve or trade to have some left over approvals (for example 5 USDT)
4. Open Swap orders page
5. Enter an amount higher than the approval amount (for example 10 USDT)
6. Press on the Approve button
7. Try closing the Reset modal

**AR**: impossible (critical for mobile devices, especially when connection issue happens there: a user can only  reload the page in order to close the modal)

**ER**: should be possible to close this

### PR DESCRIPTION
**Pull request by tukantje**
# Summary

Fixes #2522

Missing explicit modal close logic. Added a state variable to also check for to not show modal when not desired.

# To Test

1. Go to CoW Swap on Goerli,
2. Use USDT https://goerli.etherscan.io/address/0x7b77F953e703E80CD97F6911385c0b1ceabC96Bc
3. Approve or trade to have some left over approvals (for example 5 USDT)
4. Open Swap orders page
5. Enter an amount higher than the approval amount (for example 10 USDT)
6. Press on the Approve button
7. Try closing the Reset modal
8. It should be possible.

### CODE REVIEW

**Review by anxolin — APPROVED** [review]

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — COMMENTED** [review]
Hey @tukantje , once I cloe the Reset approval modal, I don't see it again when I run the 2nd round of transaction. 
I mean,
1.  close the Reset modal, 
2. Reject the Reset TX in MM
3. Press on the Approve button once again
Screenshot_8

Reproducible on Swap and Limit pages

**Comment by tukantje** [comment]
> Hey @tukantje , once I cloe the Reset approval modal, I don't see it again when I run the 2nd round of transaction. I mean,
> 
>     1. close the Reset modal,
> 
>     2. Reject the Reset TX in MM
> 
>     3. Press on the Approve button once again
>        Screenshot_8
> 
> 
> Reproducible on Swap and Limit pages

Good point. 

> Hey @tukantje , once I cloe the Reset approval modal, I don't see it again when I run the 2nd round of transaction. I mean,
> 
>     1. close the Reset modal,
> 
>     2. Reject the Reset TX in MM
> 
>     3. Press on the Approve button once again
>        Screenshot_8
> 
> 
> Reproducible on Swap and Limit pages

Good point. Fixed it, it should deploy in a minute or two.

**Review by elena-zh — APPROVED** [review]
LGTM now!

**Pull request merged**
